test/vm-download

```diff
@@ -18,78 +18,45 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
-SELF=vm-download
-. ./testlib.sh
-
-URL=http://files.cockpit-project.org/testdata/images
-
-TEST_DATA=${TEST_DATA:-$PWD}
-
-set -euf
-
-usage()
-{
-    echo >&2 "usage: $SELF [-f FLAVOR]"
-}
-
-flavor=cockpit
-
-case ${1:-} in
--f|--flavor)
-	if [ $# -lt 2 ]; then
-		usage
-		exit 2
-	fi
-	flavor=$2
-	shift 2
-	;;
---flavor=*)
-	flavor=${1#--machine=}
-	shift
-	;;
---help|-h)
-	usage
-	exit 0
-	;;
-esac
-
-os=$(python -c "print eval(open('guest/$flavor.conf', 'r').read())['os']" 2>/dev/null || true)
-os=${os:-$TEST_OS}
-
-tag=$(python -c "print eval(open('guest/$flavor.conf', 'r').read())['tags']['$os']" 2>/dev/null || true)
-tag=${tag:-0}
-
-TEST_IMAGE=$flavor-$os-$TEST_ARCH-$tag
-
-checksum_file=$TEST_IMAGE-checksum
-
-if [ ! -e "$TEST_DATA/images" ]; then
-    mkdir "$TEST_DATA/images"
-fi
-
-cd "$TEST_DATA/images"
-
-if ! curl -f -s "$URL/$checksum_file" -o "$checksum_file.remote"; then
-    if [ ! -f "$checksum_file" ]; then
-        echo Downloading $URL/$checksum_file failed.
-        exit 1
-    else
-        echo Downloading $URL/$checksum_file failed, using existing image.
-        exit 0
-    fi
-fi
-
-if [ -f "$checksum_file" ] && cmp "$checksum_file.remote" "$checksum_file"; then
-    rm -f "$checksum_file.remote"
-    exit 0
-fi
-
-cat "$checksum_file.remote" | while read sum file; do
-  if [ ! -f "$file" ] || ! echo $sum "$file" | sha256sum --status --check; then
-      echo Downloading $file
-      curl "$URL/$file.xz" | xz -d >"$file.partial" && mv "$file.partial" "$file"
-  fi
-done
-
-sha256sum --strict --quiet --check "$checksum_file.remote"
-mv "$checksum_file.remote" "$checksum_file"
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+parser = argparse.ArgumentParser(description='Get a virtual machine image')
+parser.add_argument('-v', '--verbose', action='store_true', help='Display verbose details')
+parser.add_argument('-f', '--flavor', action='store', default='cockpit', help='The flavor to download')
+args = parser.parse_args()
+
+topdir = os.path.normpath(os.path.dirname(__file__))
+arch = os.getenv("TEST_ARCH", "x86_64")
+system = os.getenv("TEST_OS", "fedora-22")
+images = os.path.join(os.getenv("TEST_DATA", topdir), "images")
+cache = os.path.join(images, "cache")
+
+with open(os.path.join(topdir, "guest", args.flavor + ".conf")) as fp:
+    config = json.load(fp)
+    revision = config.get('tags', { }).get(system, '0')
+    image = args.flavor + "-" + system + "-" + arch + "-" + revision
+    key = args.flavor + "-" + system + "-" + revision
+
+if not os.path.exists(cache):
+    os.makedirs(cache)
+
+cmd = [
+    "virt-builder",
+    "--cache", cache,
+    "--no-check-signature", # Unfortunately, not ready for this yet
+    "--arch=x86_64",
+    "--format=qcow2",
+    "--output", os.path.join(images, image),
+    "--root-password=password:foobar",
```

Comment:
I see... isn't this an indication that virt-builder is doing too much, more than we want from it?